### PR TITLE
Fix: hb_report: disable dump all tasks stack into dmesg(bsc#1158060)

### PR DESCRIPTION
**Pull request by liangxin1300**
We ever added this line to collect all process back-traces for helping to analyze a ocfs2 cluster hang problem,but in some actual production environments, there are lots of processes in a cluster node,
this dump operation will take too much time, even bring a short system hang.
Then, we should delete this operation during a generic cluster information collection.

### CODE REVIEW

**Comment by zzhou1** [comment]
Make sense to drop the code to dump all processes. Well, the code can be improved to dump the kernel stack(/proc/$(pid)/stack) of hang processes in "D" state, and even more precisely for the those processes accesing the ocfs2 mount point.

**Comment by liangxin1300** [comment]
@diegoakechi Please review, thanks

**Comment by zzhou1** [comment]
I propose the follow approach to be converted into hb_report.

PIDS=$(ps aux | awk '$8 ~ /^D/{print $2}' | tr '\n' ' ')
echo "PIDS = $PIDS"
for pid in $PIDS; do
    echo "PID = $pid"
    [ "${pid}_x" = "_x" ] || cat /proc/"$pid"/stack
done

SAMPLE OUTPUT
===========================
PIDS = 301 363 2920 3374 3386 
PID = 301
[<ffffffffc04d6bcd>] write_all_supers+0x95d/0xaf0 [btrfs]
[<ffffffffc04dde48>] btrfs_commit_transaction+0x5d8/0x8f0 [btrfs]
[<ffffffffc04d8758>] transaction_kthread+0x188/0x1b0 [btrfs]
[<ffffffffb50abea3>] kthread+0x113/0x130
[<ffffffffb5800235>] ret_from_fork+0x35/0x40
[<ffffffffffffffff>] 0xffffffffffffffff
PID = 363
[<ffffffffc0525994>] btrfs_log_inode_parent+0x4d4/0xd00 [btrfs]
[<ffffffffc05271e8>] btrfs_log_dentry_safe+0x58/0x80 [btrfs]
[<ffffffffc04f7a25>] btrfs_sync_file+0x2f5/0x440 [btrfs]
[<ffffffffb5290c58>] do_fsync+0x38/0x60
[<ffffffffb5290eac>] SyS_fsync+0xc/0x10
[<ffffffffb5003ae4>] do_syscall_64+0x74/0x150
[<ffffffffb580009a>] entry_SYSCALL_64_after_hwframe+0x3d/0xa2
[<ffffffffffffffff>] 0xffffffffffffffff
PID = 2920
[<ffffffffb50b81b2>] io_schedule+0x12/0x40
[<ffffffffc08ddf09>] jbd2_journal_commit_transaction+0xfd9/0x1700 [jbd2]
[<ffffffffc08e24d7>] kjournald2+0xb7/0x240 [jbd2]
[<ffffffffb50abea3>] kthread+0x113/0x130
[<ffffffffb5800235>] ret_from_fork+0x35/0x40
[<ffffffffffffffff>] 0xffffffffffffffff
PID = 3374
[<ffffffffb51d33a0>] balance_dirty_pages+0x270/0xd70
[<ffffffffb51d4162>] balance_dirty_pages_ratelimited+0x2c2/0x3a0
[<ffffffffb51c3edb>] generic_perform_write+0x11b/0x190
[<ffffffffb51c6c42>] __generic_file_write_iter+0x192/0x1c0
[<ffffffffc0978103>] ocfs2_file_write_iter+0x4f3/0xb10 [ocfs2]
[<ffffffffb525b44e>] __vfs_write+0xde/0x150
[<ffffffffb525bced>] vfs_write+0xad/0x1a0
[<ffffffffb525d1d2>] SyS_write+0x42/0x90
[<ffffffffb5003ae4>] do_syscall_64+0x74/0x150
[<ffffffffb580009a>] entry_SYSCALL_64_after_hwframe+0x3d/0xa2
[<ffffffffffffffff>] 0xffffffffffffffff
PID = 3386
[<ffffffffb50b81b2>] io_schedule+0x12/0x40
[<ffffffffb51c717f>] generic_file_read_iter+0x32f/0xa90
[<ffffffffb525a6bb>] __vfs_read+0xdb/0x140
[<ffffffffb525bb99>] vfs_read+0x89/0x130
[<ffffffffb525d142>] SyS_read+0x42/0x90
[<ffffffffb5003ae4>] do_syscall_64+0x74/0x150
[<ffffffffb580009a>] entry_SYSCALL_64_after_hwframe+0x3d/0xa2
[<ffffffffffffffff>] 0xffffffffffffffff

**Comment by liangxin1300** [comment]
> I propose the follow approach to be converted into hb_report.
> 
> PIDS=$(ps aux | awk '$8 ~ /^D/{print $2}' | tr '\n' ' ')
> echo "PIDS = $PIDS"
> for pid in $PIDS; do
> echo "PID = $pid"
> [ "${pid}_x" = "_x" ] || cat /proc/"$pid"/stack
> done
> 
Make sense! I will convert it to python

File: hb_report/utillib.py [hunk 1]
```diff
@@ -1649,11 +1649,29 @@ def time_status():
...
+    '''
+    out_string = ""
+    _, out, _ = crmutils.get_stdout_stderr("ps aux|awk '$8 ~ /^D/{print $2}'")
+    len_D_process = len(out.split('\n')) if out else 0
+    out_string += "Dump D-state process stack: {}\n".format(len_D_process)
+    if len_D_process == 0:
+        return out_string
+    for pid in out.split('\n'):
+        _, cmd_out, _ = crmutils.get_stdout_stderr("cat /proc/{}/comm".format(pid))
+        out_string += "pid: {}     comm: {}\n".format(pid, cmd_out)
```
Comment (arbulu89) [inline hb_report/utillib.py]: I would implement everything with the `format` method without the `+=`, but this is OK anyway.
(`out_string = "{}pid: {}     comm: {}\n".format(out_string, pid, cmd_out)

File: hb_report/utillib.py [hunk 2]
```diff
@@ -1649,11 +1649,29 @@ def time_status():
...
 
+
+def dump_D_process():
+    '''
+    dump D-state process stack
+    '''
+    out_string = ""
+    _, out, _ = crmutils.get_stdout_stderr("ps aux|awk '$8 ~ /^D/{print $2}'")
+    len_D_process = len(out.split('\n')) if out else 0
+    out_string += "Dump D-state process stack: {}\n".format(len_D_process)
```
Comment (arbulu89) [inline hb_report/utillib.py]: Why `+=` here? `=` would be just enough, right?

File: test/unittests/test_report.py [hunk 3]
```diff
@@ -362,3 +364,30 @@ def test_ts_to_dt():
 def test_which():
     ok_(which("ls"))
```
Comment (arbulu89) [inline test/unittests/test_report.py]: I would try to remove `ok_` and use the tools coming natively in Python (or pytest).
Remeber that `nose` is deprecated and we need to go getting rid off its usage eventually

File: test/unittests/test_report.py [hunk 4]
```diff
@@ -362,3 +364,30 @@ def test_ts_to_dt():
...
     ok_(not which("llll"))
+
+
+@mock.patch('crmsh.utils.get_stdout_stderr')
+def test_dump_D_process_None(mock_get_stdout_stderr):
+    mock_get_stdout_stderr.return_value = (0, None, None)
+    eq_(hb_report.utillib.dump_D_process(), "Dump D-state process stack: 0\n")
```
Comment (arbulu89) [inline test/unittests/test_report.py]: Try to remove `eq_` and use other methods

**Pull request merged**